scripts/index.js

```diff
@@ -0,0 +1,707 @@
+    return bufferedText;
+  }
+
+  var setupSegmentMetadata = function(player) {
+    // setup segment metadata
+    var segmentMetadata = document.querySelector('#segment-metadata');
+
+    player.one('loadedmetadata', function() {
+      var tracks = player.textTracks();
+      var segmentMetadataTrack;
+
+      for (var i = 0; i < tracks.length; i++) {
+        if (tracks[i].label === 'segment-metadata') {
+          segmentMetadataTrack = tracks[i];
+        }
+      }
+
+      while (segmentMetadata.children.length) {
+        segmentMetadata.removeChild(segmentMetadata.firstChild);
+      }
+
+      if (segmentMetadataTrack) {
+        segmentMetadataTrack.addEventListener('cuechange', function() {
+          var cues = segmentMetadataTrack.activeCues || [];
+
+          while (segmentMetadata.children.length) {
+            segmentMetadata.removeChild(segmentMetadata.firstChild);
+          }
+
+          for (var j = 0; j < cues.length; j++) {
+            var text = JSON.stringify(JSON.parse(cues[j].text), null, 2);
+            var li = document.createElement('li');
+            var pre = document.createElement('pre');
+
+            pre.classList.add('border', 'rounded', 'p-2');
+            pre.textContent = text;
+            li.appendChild(pre);
+            segmentMetadata.appendChild(li);
+          }
+        });
+      }
+    });
+  };
+
+  var setupPlayerStats = function(player) {
+    var currentTimeStat = document.querySelector('.current-time-stat');
+    var bufferedStat = document.querySelector('.buffered-stat');
+    var videoBufferedStat = document.querySelector('.video-buffered-stat');
+    var audioBufferedStat = document.querySelector('.audio-buffered-stat');
+    var seekableStartStat = document.querySelector('.seekable-start-stat');
+    var seekableEndStat = document.querySelector('.seekable-end-stat');
+    var videoBitrateState = document.querySelector('.video-bitrate-stat');
+    var measuredBitrateStat = document.querySelector('.measured-bitrate-stat');
+    var videoTimestampOffset = document.querySelector('.video-timestampoffset');
+    var audioTimestampOffset = document.querySelector('.audio-timestampoffset');
+
+    player.on('timeupdate', function() {
+      currentTimeStat.textContent = player.currentTime().toFixed(1);
+    });
+
+    window.statsTimer = window.setInterval(function() {
+      var oldStart;
+      var oldEnd;
+      var seekable = player.seekable();
+
+      if (seekable && seekable.length) {
+
+        oldStart = seekableStartStat.textContent;
+        if (seekable.start(0).toFixed(1) !== oldStart) {
+          seekableStartStat.textContent = seekable.start(0).toFixed(1);
+        }
+        oldEnd = seekableEndStat.textContent;
+        if (seekable.end(0).toFixed(1) !== oldEnd) {
+          seekableEndStat.textContent = seekable.end(0).toFixed(1);
+        }
+      }
+
+      // buffered
+      bufferedStat.textContent = getBuffered(player.buffered());
+
+      // exit early if no VHS
```

Comment:
new piece to make sure that we can play non HLS/DASH content here. Otherwise, the below pieces will break.